test/util.ts

```diff
@@ -330,21 +333,41 @@ export abstract class TestBuilder {
         return this;
     }
 
-    public expectToHaveDiagnostics(): this {
-        expect(this.getLuaDiagnostics()).toHaveDiagnostics();
+    public expectToHaveDiagnostic(matcher: DiagnosticMatcher): this {
+        expect(this.getLuaDiagnostics().find(matcher)).toBeDefined();
```

Comment:
Is that really something we'd need that often to add a new method there?

Reply:
We don't really have an alternative if we don't want to specify the exact diagnostic each time (which looks really messy and is not really required imo).

Reply:
I'd rather use diagnostic codes for that. In the future we should probably use an enum, or copy TypeScript's diagnostic messages concept to avoid magic numbers.